src/NzbDrone.Core/Download/Clients/Porla/Porla.cs

```diff
@@ -0,0 +1,281 @@
+                     IBlocklistService blocklistService,
+                     Logger logger)
+            : base(torrentFileInfoReader, httpClient, configService, diskProvider, remotePathMappingService, localizationService, blocklistService, logger)
+        {
+            _proxy = proxy;
+        }
+
+        public override string Name => "Porla";
+
+        public override IEnumerable<DownloadClientItem> GetItems()
+        {
+            var plist = _proxy.ListTorrents(Settings);
+
+            var items = new List<DownloadClientItem>();
+
+            // should probs paginate instead of cheating
+            foreach (var torrent in plist)
+            {
+                // we don't need to check the category, the filter did that for us, but we are checking anyway becuase why not :)
+                if (torrent.Category != Settings.Category)
+                {
+                    // TODO: Figure out how to make the test work with warnings
+                    _logger.Info($"Porla Should not have sent us a torrrent in the catagory {torrent.Category}! We expected {Settings.Category}");
+                    continue;
+                }
+
+                var outputPath = _remotePathMappingService.RemapRemoteToLocal(Settings.Host, new OsPath(torrent.SavePath));
+
+                var item = new DownloadClientItem
+                {
+                    DownloadClientInfo = DownloadClientItemClientInfo.FromDownloadClient(this, false),
+                    DownloadId = torrent.InfoHash.Hash,
+                    OutputPath = outputPath + torrent.Name,
+                    RemainingSize = torrent.Total,
+                    RemainingTime = torrent.ETA < 1 ? (TimeSpan?)null : TimeSpan.FromSeconds((double)torrent.ETA), // LibTorent uses `-1` to denote "infinite" time i.e. I am stuck, or I am done. FromSeconds will convert `-1` to 1, so we need to do this.
+                    Title = torrent.Name,
+                    TotalSize = torrent.Size,
+                    SeedRatio = torrent.Ratio
+                };
+
+                // deal with moving_storage=true ?
+
+                if (!string.IsNullOrEmpty(torrent.Error))
+                {
+                    item.Status = DownloadItemStatus.Warning;
+                    item.Message = torrent.Error;
+                } // TODO: check paused or finished first?
+                else if (torrent.FinishedDuration > 0)
+                {
+                    item.Status = DownloadItemStatus.Completed;
+                    if (torrent.ETA < 1)
+                    {
+                        // Sonarr wants to see a TimeSpan.Zero when it is done ( -1 -> 0 )
+                        item.RemainingTime = TimeSpan.Zero;
+                    }
+                }
+                else if (torrent.Flags.Contains("paused"))
+                {
+                   item.Status = DownloadItemStatus.Paused;
+                } /*  I don't know what the torent looks like if it is "Queued"
```

Comment:
Does Porla provide a list of flags we can use to determine that?

Reply:
Using porla for a bit, I actually got some examples, I'll be able to fill these in.

No. Yes? Kinda.
Libtorrent states do not match 1:1 with Sonarr states.
It's generally more "programy" and less "usery" if that makes sense.
I needed to "interpret" the logical state of a torrent by using stuff like it's time active, if it downloaded anything, etc.